src/flatc.cpp

```diff
@@ -647,20 +655,19 @@ FlatCOptions FlatCompiler::ParseFromCommandLineArguments(int argc,
       } else if (arg == "--no-leak-private-annotation") {
         opts.no_leak_private_annotations = true;
       } else if (arg == "--annotate-sparse-vectors") {
-        options.annotate_include_vector_contents = false;      
+        options.annotate_include_vector_contents = false;
       } else if (arg == "--annotate") {
         if (++argi >= argc) Error("missing path following: " + arg, true);
         options.annotate_schema = flatbuffers::PosixPath(argv[argi]);
       } else {
         // Look up if the command line argument refers to a code generator.
-        auto code_generator_it = code_generators_.find(arg);
-        if (code_generator_it == code_generators_.end()) {
+        if (!IsValidCodeGenerator(arg)) {
```

Comment:
Is there a reason to have `IsValidCodeGenerator`? Its very simple and with it, you have to do the lookup twice. Here and then in line 670 below.

Reply:
Mostly to share between this part and the part when we parse proto